autotest/cpp/test_alg.cpp

```diff
@@ -273,4 +275,125 @@ TEST_F(test_alg, GDALAutoCreateWarpedVRT_alpha_band)
     GDALClose(hWarpedVRT);
 }
 
-}  // namespace
+// Test GDALIsLineOfSightVisible() reject null dataset
+TEST_F(test_alg, GDALIsLineOfSightVisible_null_dataset)
+{
+    EXPECT_FALSE(
+        GDALIsLineOfSightVisible(nullptr, 0, 0, 0.0, 0, 0, 0.0, nullptr));
+}
+
+// Test GDALIsLineOfSightVisible() with single point dataset
+TEST_F(test_alg, GDALIsLineOfSightVisible_single_point_dataset)
+{
+    auto const sizeX = 1;
+    auto const sizeY = 1;
+    auto const numBands = 1;
+    GDALDatasetUniquePtr poDS(
+        GDALDriver::FromHandle(GDALGetDriverByName("MEM"))
+            ->Create("", sizeX, sizeY, numBands, GDT_Int8, nullptr));
+    ASSERT_TRUE(poDS != nullptr);
+
+    uint8_t val = 42;
```

Comment:
to be consistent with GDT_Int8
```suggestion
    int8_t val = 42;
```